### PR TITLE
Update the README

### PR DESCRIPTION
**Pull request by shym**
This PR brings a couple of updates to the README. In particular it lists the components as that part of the documentation is out-of-sync after #134 (sorry about that).

It is split in 2 commits, as the first commit contains modifications that are also relevant for the `4.14` branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,11 +46,12 @@ In `PREFIX/solo5-sysroot/include/nolibc`:
 In `PREFIX/lib/findlib.conf.d`:
 
 - `solo5.conf`: ocamlfind definition of the cross-compilation switch.
+- `solo5.conf`: ocamlfind definition of the cross-compilation toolchain.
```
Comment (hannesm) [inline README.md:57]: within this commit, please remove the above line - so we only have `solo5.conf` once.

**Comment by hannesm** [comment]
for the README.md on the main branch, can we as well describe that it targets OCaml 5 (with the 1.x releases) -- and we have the branch "4.14" for the OCaml 4.14 series (which is well be maintained, with 0.8.x releases)

**Comment by hannesm** [comment]
I forgot to say: thanks for your PR :)

File: README.md (hunk 1 above)
Comment (shym) [reply README.md:64]: Sorry about that and thanks. Fixed.

**Comment by shym** [comment]
> for the README.md on the main branch, can we as well describe that it targets OCaml 5 (with the 1.x releases) -- and we have the branch "4.14" for the OCaml 4.14 series (which is well be maintained, with 0.8.x releases)

Good idea. I propose that in the last commit. I think it makes more sense to add that information near the beginning of the README and so I wondered whether I should pull the small “Supported compilers” section there also.

**Pull request merged**